docs/set-up-from-template.rst

```diff
@@ -0,0 +1,184 @@
+Since the template isn't available on the ``master`` branch of the templates repository yet, you'll need to clone the repository first:
+
+.. prompt:: bash
+
+   git clone --single-branch --branch tickets/DM-23356 https://github.com/lsst/templates
+   templatekit -r templates make roundtable_aiohttp_bot
+
+Answer the prompts, and move into that directory in your shell.
+
+The rest of this tutorial uses ``safirdemo`` as the repository (and package) name:
+
+.. prompt:: bash
+
+   cd safirdemo
+
+3. Initialize the repository
+============================
+
+From the root of the project directory, initialize the Git repository:
+
+.. prompt:: bash
+
+   git init .
+   git add .
+   git commit
+
+4. Initialize the dependencies
+==============================
+
+Create a new Python virtual environment for application development (don't reuse the existing environment with templatekit).
+This is important so that your dependencies resolve reliably.
+
+Now run the initialization command:
+
+.. prompt:: bash
+
+   make update
+
+This command does several important things for you:
+
+1. Compiles the direct dependencies listed in :file:`requirements/*.in` into resolved dependencies in :file:`requirements/*.txt` files.
+2. Installs your project in a locally editable mode, along with the third-party dependencies listed in the :file:`requirements/*.txt` files.
+3. Installs tox_
+4. Installs pre-commit_ hooks
+
+After the requirements are resolved, commit those files:
+
+.. prompt:: bash
+
+   git add requirements/*.txt
+   git commit
+
+.. note::
+
+   In the future you can update your project's dependencies by re-running ``make update`` and re-committing the requirements files.
+
+   To install the project for development *without* updating dependencies, run:
+
+   .. prompt:: bash
+
+      make init
+
+5. Format the code with Black
+=============================
+
+The Python code generated by the template is good, but there may be minor formatting issues related to line length and your application's chosen name.
+You can format the code and by running tox_:
+
+.. prompt:: bash
+
+   tox -e lint
+   git commit -a
+
+6. Push to GitHub
+=================
+
+Now `create your application's repository on GitHub <https://help.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-new-repository>`__ and push to it.
+
+7. Configure Docker Hub credentials
+===================================
+
+The first push to GitHub will fail.
+That's because the Docker build step doesn't credentials for Docker Hub.
+
+To set those credentials, follow GitHub's help page `Creating and storing encrypted secrets <https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets>`__.
+The variables are:
+
+``DOCKER_USERNAME``
+    A Docker Hub username that has access to the lsstsqre organization on Docker Hub
+
+``DOCKER_PASSWORD``
+    The password correspond to ``DOCKER_USERNAME``.
```

Comment:
We should not recommend storing one's username and password in GitHub. Instead, create a Docker Hub token specific to this application and then store that in GitHub. Unfortunately, Docker Hub has no way to scope tokens, so the token will still have the ability to upload to any repository one has access to, but (a) hopefully they'll add that in the future, and (b) at least we can still revoke the tokens separately.

It's kind of painful that we'll have a proliferation of these tokens owned by random individuals who may leave the project, delete their Docker Hub accounts, or otherwise invalidate the tokens. But that part I think we can live with for now and think about later.

It looks like the Docker integration will come with a later commit. When you do finish that, I think it should assume a `DOCKER_TOKEN` secret and pass in the appropriate things to Docker to use the token. (I think you use some magic string as the username and the token as the password, but I'd have to check the docs again.)

Reply:
Good point. I'll change this and the template to refer to personal access tokens: https://docs.docker.com/docker-hub/access-tokens/

The docs show that you still use your regular username.

Reply:
Oh, okay, that's perfect then, and it makes sense to keep the username as a secret so that both are stored together. Looks great, thank you!